### ISSUE TITLE
Hello, my experiment can be run, but the results of the experiment are not quite right, how can the F1 score be 0.Can you help me? Thank you.

### ISSUE THREAD

**Issue by yao90315** [wanghr-git/TransUnetCD#2]
C:\ProgramData\Anaconda3\envs\DMINet\python.exe H:/Test/TransUnetCD/train.py
cuda
Loaded 7120 images
Loaded 1024 images
C:\ProgramData\Anaconda3\envs\DMINet\lib\site-packages\torch\functional.py:507: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at C:\cb\pytorch_1000000000000\work\aten\src\ATen\native\TensorShape.cpp:3550.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
C:\ProgramData\Anaconda3\envs\DMINet\lib\site-packages\torch\optim\lr_scheduler.py:28: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn("The verbose parameter is deprecated. Please use get_last_lr() "
Epoch 1/100:   0%|          | 0/890 [00:00<?, ?img/s]cuda
Epoch 1/100: 7120img [43:25,  2.73img/s, loss (batch)=0]
Epoch 1/100:   0%|          | 0/1024 [00:00<?, ?img/s]cuda
Epoch 1/100: 100%|██████████| 1024/1024 [05:36<00:00,  3.04img/s, loss (batch)=0]

Epoch: 000
Training:   Loss: 0.0000;
            Accuracy: 95.1198%;
Validation: Loss: 0.0000;
            Accuracy: 97.7059%;
            F1-Score: 0.0000;
Time Spent: 2942.2778s
Best F1-Score for validation : 0.0000 at epoch 000

**Comment by WangHrYii** [comment]
> C:\ProgramData\Anaconda3\envs\DMINet\python.exe H:/Test/TransUnetCD/train.py cuda Loaded 7120 images Loaded 1024 images C:\ProgramData\Anaconda3\envs\DMINet\lib\site-packages\torch\functional.py:507: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at C:\cb\pytorch_1000000000000\work\aten\src\ATen\native\TensorShape.cpp:3550.) return _VF.meshgrid(tensors, **kwargs) # type: ignore[attr-defined] C:\ProgramData\Anaconda3\envs\DMINet\lib\site-packages\torch\optim\lr_scheduler.py:28: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate. warnings.warn("The verbose parameter is deprecated. Please use get_last_lr() " Epoch 1/100: 0%| | 0/890 [00:00<?, ?img/s]cuda Epoch 1/100: 7120img [43:25, 2.73img/s, loss (batch)=0] Epoch 1/100: 0%| | 0/1024 [00:00<?, ?img/s]cuda Epoch 1/100: 100%|██████████| 1024/1024 [05:36<00:00, 3.04img/s, loss (batch)=0]
> 
> Epoch: 000 Training: Loss: 0.0000; Accuracy: 95.1198%; Validation: Loss: 0.0000; Accuracy: 97.7059%; F1-Score: 0.0000; Time Spent: 2942.2778s Best F1-Score for validation : 0.0000 at epoch 000

This code cannot be run directly, and some places need to be corrected. I will complete the code correction and organization in the near future
